src/store.cc

```diff
@@ -1129,20 +1130,9 @@ StoreEntry::abort()
 
     /* Notify the server side */
 
-    /*
-     * DPW 2007-05-07
-     * Should we check abort.data for validity?
-     */
-    if (mem_obj->abort.callback) {
-        if (!cbdataReferenceValid(mem_obj->abort.data))
-            debugs(20, DBG_IMPORTANT,HERE << "queueing event when abort.data is not valid");
-        eventAdd("mem_obj->abort.callback",
-                 mem_obj->abort.callback,
-                 mem_obj->abort.data,
-                 0.0,
-                 true);
-        unregisterAbort();
-    }
+    if (mem_obj->abortCallback && !mem_obj->abortCallback->canceled())
+        ScheduleCallHere(mem_obj->abortCallback);
+    mem_obj->abortCallback = nullptr;
```

Comment:
We should not check whether the callback was canceled because most still-set callbacks are not canceled and there are other reasons why a callback may not fire. Let ScheduleCallHere() figure it all out. The extra "will not call" debugging line may also be helpful to see when the canceled call would have happened.

```suggestion
    if (mem_obj->abortCallback) {
        ScheduleCallHere(mem_obj->abortCallback);
        mem_obj->abortCallback = nullptr;
    }
```